sycl/doc/extensions/experimental/sycl_ext_oneapi_annotated_ptr.asciidoc

```diff
@@ -524,10 +544,10 @@ class annotated_ref {
   public:
     annotated_ref(const annotated_ref&) = delete;
     operator T() const;
-    T operator=(const T &) const;
+    T operator=(T) const;
     T operator=(const annotated_ref&) const;
     // OP is: +=, -=, *=, /=, %=, <<=, >>=, &=, |=, ^=
-    T operatorOP(const T &) const;
+    T operatorOP(T) const;
```

Comment:
What is the benefit to making this change? Is it not enough to just require trivial copyability and enforce it as a static check?

Reply:
to support volatile which can't be passed by `const&`

Reply:
The benefits it's to fix the volatile type issue.

i.e.
```
void func(const volatile int& a) { ... }
int b = 1;
func(b);  // this error out
```
with 
```
void func(volatile int a) { ... }
int b = 1;
func(b);  // this will work
```

Reply:
Thank you for clarifying. 👍